docs/notebooks/multiobjective_ehvi.pct.py

```diff
@@ -0,0 +1,176 @@
+    lengthscale = l * tf.ones(input_dim, dtype=gpflow.default_float())
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=lengthscale)
+    single_outcome_dataset = Dataset(
+        data.query_points, tf.gather(data.observations, [outcome_idx], axis=1)
+    )
+    gpr = gpflow.models.GPR(single_outcome_dataset.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+    return create_model(
+        {
+            "model": gpr,
+            "optimizer": gpflow.optimizers.Scipy(),
+            "optimizer_args": {
+                "minimize_args": {"options": dict(maxiter=100)},
+            },
+        }
+    )
+
+
+objective_models = [
+    (
+        create_single_outcome_model(
+            initial_data[OBJECTIVE],
+            outcome_idx=i,
+        ),
+        1,
+    )
+    for i in range(num_objective)
+]
+
+# And stack the two independent GP model in a single `ModelStack` multioutput model representing predictions through different outputs.
+
+models = {OBJECTIVE: ModelStack(*objective_models)}
+
+# ## Define the acquisition function
+# Here we utilize the [EHVI](https://link.springer.com/article/10.1007/s10898-019-00798-7): `ExpectedHypervolumeImprovement` acquisition function:
+
+from trieste.acquisition.rule import EfficientGlobalOptimization
+
+ehvi = ExpectedHypervolumeImprovement()
+rule: EfficientGlobalOptimization[Box] = EfficientGlobalOptimization(builder=ehvi.using(OBJECTIVE))
+
+# ## Run the optimization loop
+#
+# We can now run the optimization loop
+
+num_steps = 20
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+result = bo.optimize(num_steps, initial_data, models, acquisition_rule=rule)
+
+# To conclude, we visualize the queried data across the design space.
+
+# +
+datasets = result.try_get_final_datasets()
+data_query_points = datasets[OBJECTIVE].query_points.numpy()
+data_observations = datasets[OBJECTIVE].observations.numpy()
+
+_, ax = plot_function_2d(
+    vlmop2, mins, maxs, grid_density=100, contour=True, title=["Obj 1", "Obj 2"]
+)
+plot_bo_points(data_query_points, ax=ax[0, 0], num_init=num_initial_points)
+plot_bo_points(data_query_points, ax=ax[0, 1], num_init=num_initial_points)
+plt.show()
+# -
+
+# ... and visulize in the objective space, purple dots denotes the nondominated points.
+
+plot_mobo_points_in_obj_space(data_observations, num_init=num_initial_points)
+plt.show()
+
+# We can also visualize how the performance metric get envolved with respect to the number of BO iterations.  
+# First, we need to define a performance metric. Many metrics have been considered for multi-objective function. Here, we use the log hypervolume difference.
+
+# The log hypervolume difference is defined as the difference between the hypervolume of the true Pareto front and the hypervolume of the approximate Pareto front based on the observed data.
+# $$
```

Comment:
Many metrics have been considered for multi-objective *optimization*.

Reply:
Here, we use the log hypervolume difference, defined as the difference between the hypervolume of the true Pareto front and the hypervolume of .....